transport_nantes/mobilito/models.py

```diff
@@ -13,6 +13,19 @@ class MobilitoUser(models.Model):
     """Maintain information about Mobilito users."""
 
     user = models.ForeignKey(User, on_delete=models.PROTECT, related_name="+")
+    # null or user defined
+    _username = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        unique=True,
+        verbose_name="Nom d'utilisateur",
+    )
+    # A sha1 identifier to display in URLs
+    sha1_identifier = models.CharField(
```

Comment:
Somehow we got in the habit of using SHA1 to generate unique random keys rather than just generating random strings.  I think we should get ourselves out of this habit (and rename the fields to something more logical, like "human_identifier".

First, there's no benefit to sha1 here over just a random string.

Second, the fact that it's called sha1 suggests that we've actually hashed something known to arrive at the value, and that's not at all what we're doing.  We never start with a value to get the hash to find something, in the sense of a hash table.  So we are presenting misleading information in the code.

Finally, since SHA1 is known to suffer from collision attacks, a reader could well wonder why we haven't migrated to a safer hash algorithm, like SHA256 or SHA512 (or at least SHA1DC if we were bound, git-style, to SHA1-style hashes).  The reason we don't care is that we aren't using the fact that it's a cryptographic hash function, and that's likely quite confusing.

Reply:
Understood. Do you want me to redo migrations and use django.crypto to generate a random string instead in this PR ?

Reply:
No, the values are fine.  I'm thinking we just accidentally got in a bad habit and should correct the field names and the method of generation.

From there, it doesn't matter that some are not pseudo-random.  They are unique strings with no clear sense to them.  That's enough.